libraries/Microsoft.Bot.Connector.Streaming/Transport/NamedPipeTransport.cs

```diff
@@ -152,6 +134,93 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        private async Task RetryToSucceedAsync(Func<CancellationToken, Task> createNamedPipeFactory, CancellationToken cancellationToken)
+        {
+            _tryTimes = 0;
+
+            // TryTimes exhausted happens when customers use the app.UsedNamedPipe in their bot, but do not enable ase.
+            while (!cancellationToken.IsCancellationRequested && _tryTimes > 100)
+            {
+                _tryTimes++;
+
+                // To avoid NamedPipeServer and NamedPipeClient dead lock, use random backoff when create NamePipe
+                using (var source = new CancellationTokenSource())
+                {
+                    try
+                    {
+                        // NamedPipeServer and NamedPipeClient dead lock can be break by time out
+                        // Add this random backoff to avoid a special dead lock when machine run slow.
+                        // At least 5 seconds, because :
+                        // If ASE update to new version but customer still use old version bot(SDK). Quick retry will make this connection never connect.
+                        var timeoutTask = Task.Delay(TimeSpan.FromSeconds((new Random(DateTime.Now.Millisecond).NextDouble() * _tryTimes) + 5), cancellationToken);
```

Comment:
Do we need `new Random` every time?

Since `_tryTimes <= 100`, it will toward to 50 seconds in avg, is it expected?